src/interaction.js

```diff
@@ -176,7 +176,7 @@ export class KeystoreInteraction {
    * supported.  See the Bowser documentation for more details:
    * https://github.com/lancedikson/bowser
    *
-   * @returns {boolean} whether this interaction is supported
+   * @returns {boolean | Promise<boolean>} whether this interaction is supported
```

Comment:
What's going on with this? Returning a union between a `Promise` type and a `boolean` seems like it has a high bug potential since they each are evaluated very differently. A `Promise` is truthy and must be waited for. If this can be async, please make it `Promise<boolean>` only and have non-promise instances `return Promise.resolve(boolean)`

Reply:
Yep. Good point. the fun challenge of being at the beginning of our TS migration journey

Reply:
So I started by making it only be a Promise. but that had a bunch of a downstream consequences of all the other places that call this (to your point about being handled differently). I opted instead to just rename the `isSupported` in the Ledger interactions that is the only one that needs to be async (because it involves an app interaction to check the version) to be `isAppSupported` and use that for this one particular use case. Makes the change much smaller: [9ddccb2](https://github.com/unchained-capital/unchained-wallets/pull/88/commits/9ddccb2fd7a2b07210178e2091fd8e29d144177f)